modules/citrus-integration/src/it/java/com/consol/citrus/javadsl/runner/GlobalVariablesTestRunnerIT.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2006-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.consol.citrus.javadsl.runner;
+
+import com.consol.citrus.annotations.CitrusResource;
+import com.consol.citrus.annotations.CitrusTest;
+import com.consol.citrus.context.TestContext;
+import com.consol.citrus.dsl.runner.TestRunner;
+import com.consol.citrus.dsl.testng.TestNGCitrusTestRunner;
+import org.testng.annotations.Optional;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+/**
+ * Tests the loading and usage of global variables in citrus
+ */
+@Test
+public class GlobalVariablesTestRunnerIT extends TestNGCitrusTestRunner {
+
+    @CitrusTest
+    public void globalProperties() {
+        echo("Project name is: ${project.name}");
+
+        echo("Testing global variables from properties: ${globalWelcomingText}");
+    }
+
+    @CitrusTest
+    @Parameters({"runner", "context"})
+    public void randomLocalVariable_readMultipleTimes_valueShouldRemainConstantOnEachRead(
+            @Optional @CitrusResource TestRunner runner, @Optional @CitrusResource TestContext context) {
+
+        runner.variable("localRandomNumber1", "citrus:randomNumber(5)");
+        runner.echo("localRandomNumber1: ${localRandomNumber1}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "localRandomNumber1");
+
+        runner.createVariable("localRandomNumber2", "citrus:randomNumber(8)");
+        runner.echo("localRandomNumber2: ${localRandomNumber2}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "localRandomNumber2");
+    }
+
+    @CitrusTest
+    @Parameters({"runner", "context"})
+    public void randomGlobalVariable_readMultipleTimes_valueShouldRemainConstantOnEachRead(
+            @Optional @CitrusResource TestRunner runner, @Optional @CitrusResource TestContext context) {
+        runner.echo("globalRandomNumber1: ${globalRandomNumber1}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "globalRandomNumber1");
+    }
+
+    @CitrusTest
+    @Parameters({"runner", "context"})
+    public void multipleGlobalVariablesWithCrossReferencesAndFunctions_readMultipleTimes_valueShouldRemainConstantOnEachRead(
+            @Optional @CitrusResource TestRunner runner, @Optional @CitrusResource TestContext context) {
+        runner.echo("globalSum1: ${globalSum1}");
+        assertVariableValuesMatchWhenReadMultipleTimes(context, "globalSum1");
```

Comment:
In my opinion, this is redundant to line [60](https://github.com/citrusframework/citrus/pull/573/files#diff-632f5ba22ee89f7d193923dd4e328331R60), because `globalSum1` is just a placeholder for a Citrus function just as `globalRandomNumber1` is. The same goes for `globalSum2`. So I think it would be enough to test for matching variables values here.

Reply:
Agreed. I can't remember why i thought this necessary at the time.